src/main/kotlin/com/zepben/evolve/database/sqlite/upgrade/changesets/ChangeSet46.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.upgrade.changesets
+
+import com.zepben.evolve.database.sqlite.upgrade.ChangeSet
+
+internal fun changeSet46() = ChangeSet(
+    46,
+    `Make potential_transformers type non-null`
+)
+
+@Suppress("ObjectPropertyName")
+private val `Make potential_transformers type non-null` = listOf(
+    """
```

Comment:
You should actually be able to do this with ALTER TABLE RENAME/ADD/DROP column (assuming we're using a recent enough version of sqlite)

ALTER TABLE RENAME COLUMN X TO Y
ALTER TABLE ADD COLUMN X ...
INSERT INTO ... (copy data from Y to X somehow - probably a smart way to do this)
ALTER TABLE DROP COLUMN Y

Reply:
Adding a non-null column to an existing table requires specifying a default. AFAIK no other non-null column has a default value.

Reply:
Maybe I can drop the default after creating it? Let me check

Reply:
You should be able to just specify the default on the ADD... I.e something like
```ADD COLUMN X NOT NULL DEFAULT 'UNKNOWN'```

Reply:
I know, but this has the unintended consequence of allowing an unspecified `type` when inserting new entries.

Reply:
So after discussion with Marcus we've decided to leave this method in place.
In summary, with the RENAME + ADD + DROP method we would introduce a side effect which in this case is not bad, but introduces inconsistency in our schema.
Notably, to add a column with `NOT NULL` requires specifying a default value for the column. We could do this and specify UNKNOWN, but then this would create the side-effect for all future insert statements that if they do not specify the type field they will get given UNKNOWN for type rather than an error.

In reality this isn't a real issue, because we always insert all fields from the SDK, but over time we'd end up with schemas with column defaults in some cases and not others. In the interest of keeping the schema consistent we'll avoid this.

A potential solution is to re-work all enum columns to have a default, but this would require updating SqliteTable and writing a lot of upgrade scripts, which we deemed low value and an unnecessarily large body of work for just a simple bug fix. Someone may or may not like to do that in the future, but there's no pressing need. In reality we'll just wait until one day ALTER COLUMN is properly supported and do things properly then.